plugins/hw_event/hw_event_plugin.go

```diff
@@ -200,10 +114,48 @@ func publishHwEvent(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	for i, e := range redfishEvent.Events {
+		if e.Message == "" {
+			parsed, err := parseMessage(e)
+			if err == nil {
+				redfishEvent.Events[i] = parsed
+			} else {
+				// ignore error
+				log.Debugf("error parsing message: %v", err)
+			}
+
+		}
+	}
+
 	data := v1hwevent.CloudNativeData()
 	data.SetVersion("v1") //nolint:errcheck
 	data.SetData(&redfishEvent)
 
-	event, _ := common.CreateHwEvent(pub.ID, "HW_EVENT", data)
-	_ = common.PublishHwEvent(scConfig, event)
+	event, _ := common.CreateHwEvent(pub.ID, "HW_EVENT", data, timeReceived)
```

Comment:
instead of passing the time created, you can set the event object after it is received.
Also check for error  before setting

Reply:
updated